x-pack/elastic-agent/pkg/agent/application/info/host_id_windows.go

```diff
@@ -0,0 +1,26 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package info
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+)
+
+var winReg = regexp.MustCompile("[\\s]+MachineGuid[\\s]+REG_SZ[\\s]+([\\S]+)")
+
+func loadHostID() (string, error) {
+	c := exec.Command("Reg", "query", "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography", "/v", "MachineGuid")
```

Comment:
Being that reading from Windows registry is pretty standard for any Windows application, I think it would be better to do this using win32 api's then calling a subcommand.

https://pkg.go.dev/golang.org/x/sys/windows/registry